src/context/script-let.ts

```diff
@@ -528,6 +528,26 @@ export class ScriptLetContext {
     this.closeScopeCallbacks.pop()!();
   }
 
+  public appendDeclareMaybeStores(maybeStores: Set<string>): void {
+    const reservedNames = new Set<string>([
+      "$$props",
+      "$$restProps",
+      "$$slots",
+    ]);
+    for (const nm of maybeStores) {
+      if (reservedNames.has(nm)) continue;
+
+      this.appendScriptWithoutOffset(
+        `declare let $${nm}: Parameters<Parameters<(typeof ${nm})["subscribe"]>[0]>[0];`,
```

Comment:
What if there is already an identifier named `$nm` or it's already declared by the user manually?

This could be uncommon, but before this PR this could happen if the user want to workaround this issue, then this PR would break that.

Reply:
I checked the edge case and it didn't seem to be a problem.
https://github.com/ota-meshi/svelte-eslint-parser/blob/store-type/tests/fixtures/parser/ast/ts-store01-type-output.svelte

Also, the svelte compiler doesn't seem to allow you to declare variables that start with $ yourself.



<img width="483" alt="image" src="https://user-images.githubusercontent.com/16508807/180797302-84f97c0d-d317-40e4-b1cd-fe1689884fa9.png">

```svelte
<script>
	let name = 'world';
	const $n = name
</script>

<h1>Hello {name}!</h1>
```

Reply:
Does `script` with `lang="ts"` + `declare const $n = xxx` works the same?

Reply:
I added a test case to check that edge case. That doesn't seem to be a problem either 😉.

https://github.com/ota-meshi/svelte-eslint-parser/blob/store-type/tests/fixtures/parser/ast/ts-store02-type-output.svelte

Reply:
Wow, that is a bit of surprise to me

https://www.typescriptlang.org/play?#code/CYUwxgNghgTiAEYD2A7AzgF3gEigLnkxgEsUBzAKFElgWXS1wKNMqA

Reply:
OK, it should due to

```ts
tokens.length = 0
comments.length = 0
removeAllScope(node, result)
```

I just learned a lot from this PR!